WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -953,14 +957,21 @@ private async Task<IEnumerable<TxOut>> ProceedWithOutputRegistrationPhaseAsync(u
 		// Calculate outputs values
 		var constructionState = roundState.Assert<ConstructionState>();
 
-		AmountDecomposer amountDecomposer = new(roundParameters.MiningFeeRate, roundParameters.AllowedOutputAmounts, Constants.P2wpkhOutputVirtualSize, Constants.P2wpkhInputVirtualSize, (int)availableVsize);
+		// Get the output's size and its of the input that will spend it in the future.
+		// Here we assume all the outputs share the same scriptpubkey type.  
```

Comment:
> We assume that all the inputs are sharing the same scriptpubkey type, right?

No.

> add scriptpubkey type for inputs to decomposer logic?

No.

> add scriptpubkey type for outputs to decomposer logic? 

Yes.

Reply:
I guess this code helps the testings when you created a clearly taproot and a clearly segwit wallet and mix with them. In the perspective of the final product, this code is wrong - because inputs and outputs can be with mixed scripts.